pennylane/qnode.py

```diff
@@ -419,6 +419,13 @@ def _validate_adjoint_method(device, interface):
                 f"The {device.short_name} device does not support adjoint differentiation."
             )
 
+        if device.shots is not None:
+            warnings.warn(
+                "Adjoint diff requested on device with finite shots. Adjoint"

```

Comment:
How come there's a warning here too? :thinking: Would it suffice to have it only on the device level?

On another note, a similar comment related to the message would be great to rephrase.

Reply:
This is when users can most easily do something about the warning.  They haven't already invested in a forward pass.